tools/testsys/src/aws_resources.rs

```diff
@@ -342,6 +341,317 @@ impl Migration for AwsK8s {
+        let start_migrate = self.migration_crd(
+            format!("{}-2-migrate", self.cluster_name()),
+            MigrationVersion::Migrated,
+            Some(depends_on.clone()),
+            testsys_images,
+        )?;
+        depends_on.push(start_migrate.name().context("Crd missing name")?);
+        let migrated =
+            self.ecs_test_crd("-3-migrated", Some(depends_on.clone()), testsys_images)?;
+        depends_on.push(migrated.name().context("Crd missing name")?);
+        let end_migrate = self.migration_crd(
+            format!("{}-4-migrate", self.cluster_name()),
+            MigrationVersion::Starting,
+            Some(depends_on.clone()),
+            testsys_images,
+        )?;
+        depends_on.push(end_migrate.name().context("Crd missing name")?);
+        let last = self.ecs_test_crd("-5-final", Some(depends_on.clone()), testsys_images)?;
+        Ok(vec![
+            ecs,
+            ec2,
+            initial,
+            start_migrate,
+            migrated,
+            end_migrate,
+            last,
+        ])
+    }
+
+    /// Labels help filter test results with `testsys status`.
+    fn labels(&self) -> BTreeMap<String, String> {
+        btreemap! {
+            "testsys/arch".to_string() => self.arch.to_string(),
+            "testsys/variant".to_string() => self.variant.to_string(),
+        }
+    }
+
+    fn kube_arch(&self) -> String {
+        self.arch.replace('_', "-")
+    }
+
+    fn kube_variant(&self) -> String {
+        self.variant.replace('.', "")
+    }
+
+    /// Bottlerocket cluster naming convention.
+    fn cluster_name(&self) -> String {
+        self.target_cluster_name
+            .clone()
+            .unwrap_or_else(|| format!("{}-{}", self.kube_arch(), self.kube_variant()))
+    }
+
+    fn ecs_crd(&self, testsys_images: &TestsysImages) -> Result<Crd> {
+        let cluster_name = self.cluster_name();
+        let ecs_crd = Resource {
+            metadata: ObjectMeta {
+                name: Some(cluster_name.clone()),
+                namespace: Some(NAMESPACE.into()),
+                labels: Some(self.labels()),
+                ..Default::default()
+            },
+            spec: ResourceSpec {
+                depends_on: None,
+                agent: Agent {
+                    name: "ecs-provider".to_string(),
+                    image: testsys_images.ecs_resource.clone(),
+                    pull_secret: testsys_images.secret.clone(),
+                    keep_running: false,
+                    timeout: None,
+                    configuration: Some(
+                        EcsClusterConfig {
+                            cluster_name,
+                            region: Some(self.region.clone()),
+                            assume_role: self.assume_role.clone(),
+                            vpc: None,
+                        }
+                        .into_map()
+                        .context("Unable to convert ecs config to map")?,
```

Comment:
Pedantic nit: `s/ecs/ECS`, or if you are referring to the structure, use the whole name `EcsClusterConfig`, the same applies to `s/ec2/EC2`